### DISCUSSION TITLE
About normalization issues on RNN

### DISCUSSION THREAD

**Discussion by parkinkon1** [seominjoon/kaist-ai605#40] [Q&A] [answered]
Hello professor,

I have a few questions of implementing network normalizations,

1. Can I use dropout within the recurrent structure of the RNN? In my opinion, the information of the hidden states will disappear while passing dropout layers, and past features may cannot remembered for a long time. Is it correct to use dropout only for input tensors and classification layers?

2. Similarly, I wonder if batch normalization can be used within the recurrent structure, or can only be used for the input tensors and classification layers.

Thank you!

**Comment by seominjoon** [comment, marked as answer]
1. You are correct, this is why we usually don't use dropout in the recurrent structure. So it is recommended to use dropout only for input or ouptut (classification layer).
2. Batch norm is not frequently used in in RNN or Transformer. LayerNorm is used instead. There are many potential explanations for why such discrepancy exists, but no one seems to know the exact reason ;) It is more of an empirical reason.

**Reply by parkinkon1** [reply]
Thank you!
